framework/web/Request.php

```diff
@@ -261,6 +261,13 @@ class Request extends \yii\base\Request
         'Front-End-Https' => ['on'], // Microsoft
     ];
 
+    /**
+     * @var mixed[] The request "attributes" can be used to for storage of any parameters derived from the request,
+     * e.g. the scopes of an Oauth request.
+     * @since 2.0.44
+     */
+    protected $_attributes = [];
```

Comment:
add `@property` tag to class

Reply:
@property-read tag added (since the attributes are not directly writable)

Reply:
@rhertogh `@property-read array $attributes` and `@property mixed $attribute`

Reply:
@WinterSilence How would `@property mixed $attribute` work? The `getAttribute($name, $default = null)` requires the `$name` parameter, is it still possible to define `attribute` as a property of the class?

Reply:
@rhertogh ups.. you right, all fine